ksql-streams/src/test/java/io/confluent/ksql/execution/streams/StreamStreamJoinBuilderTest.java

```diff
@@ -0,0 +1,331 @@
+import io.confluent.ksql.execution.plan.ExecutionStep;
+import io.confluent.ksql.execution.plan.Formats;
+import io.confluent.ksql.execution.plan.JoinType;
+import io.confluent.ksql.execution.plan.StreamStreamJoin;
+import io.confluent.ksql.query.QueryId;
+import io.confluent.ksql.schema.ksql.LogicalSchema;
+import io.confluent.ksql.schema.ksql.PhysicalSchema;
+import io.confluent.ksql.schema.ksql.types.SqlTypes;
+import io.confluent.ksql.serde.Format;
+import io.confluent.ksql.serde.FormatInfo;
+import io.confluent.ksql.serde.KeyFormat;
+import io.confluent.ksql.serde.KeySerde;
+import io.confluent.ksql.serde.SerdeOption;
+import io.confluent.ksql.serde.ValueFormat;
+import java.time.Duration;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.streams.kstream.JoinWindows;
+import org.apache.kafka.streams.kstream.Joined;
+import org.apache.kafka.streams.kstream.KStream;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+public class StreamStreamJoinBuilderTest {
+  private static final String LEFT = "LEFT";
+  private static final String RIGHT = "RIGHT";
+  private static final String ALIAS = "ALIAS";
+  private static final LogicalSchema LEFT_SCHEMA = LogicalSchema.builder()
+      .valueColumn("BLUE", SqlTypes.STRING)
+      .valueColumn("GREEN", SqlTypes.INTEGER)
+      .build()
+      .withAlias(LEFT)
+      .withMetaAndKeyColsInValue();
+  private static final LogicalSchema RIGHT_SCHEMA = LogicalSchema.builder()
+      .valueColumn("RED", SqlTypes.BIGINT)
+      .valueColumn("ORANGE", SqlTypes.DOUBLE)
+      .build()
+      .withAlias(RIGHT)
+      .withMetaAndKeyColsInValue();
+  private static final LogicalSchema SCHEMA = LogicalSchema.builder()
+      .valueColumn("BLUE", SqlTypes.STRING)
+      .valueColumn("GREEN", SqlTypes.STRING)
+      .valueColumn("RED", SqlTypes.BIGINT)
+      .valueColumn("ORANGE", SqlTypes.DOUBLE)
+      .build()
+      .withAlias(ALIAS)
+      .withMetaAndKeyColsInValue();
+  private static final PhysicalSchema LEFT_PHYSICAL =
+      PhysicalSchema.from(LEFT_SCHEMA, SerdeOption.none());
+  private static final PhysicalSchema RIGHT_PHYSICAL =
+      PhysicalSchema.from(RIGHT_SCHEMA, SerdeOption.none());
+  private static final Formats LEFT_FMT = Formats.of(
+      KeyFormat.nonWindowed(FormatInfo.of(Format.KAFKA)),
+      ValueFormat.of(FormatInfo.of(Format.JSON)),
+      SerdeOption.none()
+  );
+  private static final Formats RIGHT_FMT = Formats.of(
+      KeyFormat.nonWindowed(FormatInfo.of(Format.KAFKA)),
+      ValueFormat.of(FormatInfo.of(Format.AVRO)),
+      SerdeOption.none()
+  );
+  private static final Duration BEFORE = Duration.ofMillis(1000);
+  private static final Duration AFTER = Duration.ofMillis(2000);
+  private static final JoinWindows WINDOWS = JoinWindows.of(BEFORE).after(AFTER);
+
+  @Mock
+  private KStream leftStream;
```

Comment:
This test us using raw types.  Can we please change the test to use generic types please for all these fields.

Same goes for the other tests.